app/domain/factories/create_from_policy_factory.rb

```diff
@@ -169,14 +169,16 @@ def render_and_apply_policy(policy_load_path:, policy_template:, variables:, acc
     end
 
     def set_factory_variables(schema_variables:, factory_variables:, variable_path:, authorization:, account:)
-      # Only set secrets defined in the policy
-      schema_variables.each_key do |factory_variable|
-        variable_id = @uri.encode_www_form_component("#{variable_path}/#{factory_variable}")
+      # Only set secrets defined in the policy and present in factory payload
+      (schema_variables.keys & factory_variables.keys).each do |schema_variable|
+        next unless factory_variables.key?(schema_variable)
```

Comment:
nit: shouldn't have to check that `schema_variables` gets a positive response from `factory_variables.key?` because it's baked into the parent array `(schema_variables.keys & factory_variables.keys)`.

Reply:
Good catch. This was left over after I realized finding the intersection of keys was much more effective.